server/src/main/java/org/opensearch/extensions/AddSettingsUpdateConsumerRequest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.transport.TransportRequest;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.WriteableSetting;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Add Settings Update Consumer Request for Extensibility
+ *
+ * @opensearch.internal
+ */
+public class AddSettingsUpdateConsumerRequest extends TransportRequest {
+    private final DiscoveryExtensionNode extensionNode;
+    private final List<WriteableSetting> componentSettings;
+
+    public AddSettingsUpdateConsumerRequest(DiscoveryExtensionNode extensionNode, List<Setting<?>> componentSettings) {
+        this.extensionNode = extensionNode;
+        this.componentSettings = new ArrayList<>(componentSettings.size());
+        for (Setting<?> setting : componentSettings) {
+            this.componentSettings.add(new WriteableSetting(setting));
+        }
+    }
+
+    public AddSettingsUpdateConsumerRequest(StreamInput in) throws IOException {
+        super(in);
+
+        // Set extension node to send update settings request to
+        this.extensionNode = new DiscoveryExtensionNode(in);
+
+        // Read in component setting list
+        int componentSettingsCount = in.readVInt();
+        List<WriteableSetting> componentSettings = new ArrayList<>(componentSettingsCount);
```

Comment:
Minor, we could probably use `this.componentSettings = new ArrayList<>(componentSettingsCount);` here since this is the constructor.